src/domain/auth-rebac/domain/src/services/rebac_service.rs

```diff
@@ -0,0 +1,94 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::HashMap;
+
+use opendatafabric::{AccountID, DatasetID};
+
+use crate::{
+    DeleteEntitiesRelationError,
+    DeleteEntityPropertyError,
+    GetEntityPropertiesError,
+    GetEntityRelationsError,
+    InsertEntitiesRelationError,
+    PropertyName,
+    PropertyValue,
+    Relation,
+    UpsertEntityPropertyError,
+};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait RebacService: Send + Sync {
```

Comment:
I conceived of this service to hide the more general repository interface into a more specific of the service.

For example, to avoid constantly passing `entity_type: EntityType, entity_id: EntityId`, for dataset -- to eliminate the possibility (at the interface level) of passing `EntityType::Account` and `EntityId` of dataset (`dataset_id`)

Reply:
Ah I see now. That makes sense.

Although I wonder whether we will still need access to the generic interface to connect this info to OSO.

How OSO integration will work is where I lack clarity the most, so it might be worth as one of your next steps to provide stub implementation of these services and create a test that connects data to the OSO ruleset.